### ISSUE TITLE
Feature request: Allow conversation retention & multiple conversations in `gr.ChatInterface`

### ISSUE THREAD

**Issue by jamie0725** [gradio-app/gradio#10045]
- [ X ] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  
Chatbots created using `gr.ChatInterface` currently do not retain conversation histories when the web page is closed. Each time a user closes and reopens the web page, the conversation history is lost. This can be frustrating in real-world scenarios where users may want to ask follow-up questions periodically.

Additionally, users are currently limited to creating a single conversation, which is inconvenient. Different topics may require separate conversations, and users need the ability to manage multiple conversations simultaneously.

**Describe the solution you'd like**  
It would be similar to many existing chatbot applications that retain conversation histories using front-end caching. As long as users do not clear their browser history, the conversation histories will remain intact.

![image](https://github.com/user-attachments/assets/54cdb62f-b5e3-4a24-accd-48a197e7ac5d)

**Comment by abidlabs** [comment]
Yes this would be really nice, and its on our roadmap, cc @dawoodkhan82

**Comment by brycepg** [comment]
I have an alternative implementation of this with langgraph here: https://github.com/brycepg/gradio-langgraph-template/blob/main/app_with_tabs.py
